.github/workflows/ci-cd.yml

```
@@ -0,0 +1,166 @@
+    - name: Run security audit
+      run: pnpm audit --audit-level=moderate
+      
+    - name: Run dependency check
+      run: pnpm outdated
+
+  deploy-staging:
+    name: Deploy to Staging
+    runs-on: ubuntu-latest
+    needs: [lint-and-format, test, build, security-audit]
+    if: github.ref == 'refs/heads/staging' && github.event_name == 'push'
+    
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)